src/canopy/knowledge_base/record_encoder/openai.py

```diff
@@ -27,25 +36,31 @@ def __init__(self,
                         Defaults to 400.
             **kwargs: Additional arguments to pass to the underlying `pinecone-text. OpenAIEncoder`.
         """  # noqa: E501
-        encoder = OpenAIEncoder(model_name, **kwargs)
+        try:
+            encoder = OpenAIEncoder(model_name, **kwargs)
+        except OpenAIError as e:
+            raise RuntimeError(
+                "Failed to connect to OpenAI, please make sure that the OPENAI_API_KEY "
+                "environment variable is set correctly.\n"
+                f"Error: {_format_openai_error(e)}"
+            ) from e
         super().__init__(dense_encoder=encoder, batch_size=batch_size)
 
-    def encode_documents(self, documents: List[KBDocChunk]) -> List[KBEncodedDocChunk]:
-        """
-        Encode a list of documents, takes a list of KBDocChunk and returns a list of KBEncodedDocChunk.
-
-        Args:
-            documents: A list of KBDocChunk to encode.
-
-        Returns:
-            encoded chunks: A list of KBEncodedDocChunk, with the `values` field populated by the generated embeddings vector.
-        """  # noqa: E501
-        return super().encode_documents(documents)
-
     async def _aencode_documents_batch(self,
                                        documents: List[KBDocChunk]
                                        ) -> List[KBEncodedDocChunk]:
         raise NotImplementedError
 
     async def _aencode_queries_batch(self, queries: List[Query]) -> List[KBQuery]:
         raise NotImplementedError
+
+    def _format_error(self, err):
```

Comment:
Maybe we should resuse this code in the LLM too? I think it's can improve the experience

Reply:
Yeah, that's the plan. We'll do it as part of the planned `BaseLLM` refactor anyway